### PR TITLE
879-pridat-do-txm-docker-composu-authentik

### PR DESCRIPTION
**Pull request by NotLubo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,3 +124,27 @@ To obtain configuration in the code, one should call `get_application_configurat
...
+We are using Authentik for authentication. You can find more information about it [here](https://goauthentik.io/docs).
+After you run `make docker-build` you must setup Authentik. You will do this only once. 
+It might take some time (minute or two) before Authentik is ready.
+1. Go to http://localhost:9000/if/flow/initial-setup/
+2. Pick email and password
+3. Go to Admin interface
+4. Click on Applications / Providers
+5. Click Create
+6. Select OAuth2/OpenID Provider
+7. Input TXM as name
```
Comment (kubantjan) [inline README.md:138]: where is this TXM taken from? Is it somewhere in code as well?

File: README.md [hunk 2]
```diff
@@ -124,3 +124,27 @@ To obtain configuration in the code, one should call `get_application_configurat
...
+8. Client ID: `f5c6b6a72ff4f7bbdde383a26bdac192b2200707`
```
Comment (kubantjan) [inline README.md:139]: Where is client id a nd secret taken from? from .env? Then I would not write the exact ids and secrets here, but say to take them from .env

File: README.md [hunk 3]
```diff
@@ -124,3 +124,27 @@ To obtain configuration in the code, one should call `get_application_configurat
...
+9. Client Secret: `37e841e70b842a0d1237b3f7753b5d7461307562568b5add7edcfa6630d578fdffb7ff4d5c0f845d10f8f82bc1d80cec62cb397fd48795a5b1bee6090e0fa409`
+10. Redirect URI: `http://localhost:8080/v1/user/authentik-login`
+11. Click on Applications
+12. Create new application
+13. Set name as TXM and slug as txm
```
Comment (kubantjan) [inline README.md:144]: where is this TXM and txm taken from? Is it somewhere in code as well?

File: README.md [hunk 4]
```diff
@@ -124,3 +124,27 @@ To obtain configuration in the code, one should call `get_application_configurat
...
+14. From providers select TXM
```
Comment (kubantjan) [inline README.md:145]: Where are this providers taken from?

**Review by kubantjan — CHANGES_REQUESTED** [review]

File: txmatching/web/api/user_api.py [hunk 5]
```diff
@@ -153,3 +155,31 @@ def post(self):
...
+        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
+        authentik_res = requests.post('http://host.docker.internal:9000/application/o/token/',
+                                      data=data, headers=headers)
+        authentik_res = authentik_res.json()
+
+        response = response_ok(authentik_res)
+        response.set_cookie("access_token", value=authentik_res["access_token"])
+        response.set_cookie("refresh_token", value=authentik_res["refresh_token"])
+
+        # todo redirect to homepage instead of showing access token after testing is finished
```
Comment (kubantjan) [inline txmatching/web/api/user_api.py:184]: add link to an issue

File: README.md (hunk 1 above)
Comment (NotLubo) [reply README.md:138]: Its just label used in authentic you could put anything there

**Review by kubantjan — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (NotLubo) [reply README.md:139]: Yes its from env. Corrected

File: README.md (hunk 4 above)
Comment (NotLubo) [reply README.md:145]: You create provider in steps 4 -> 10. I added subheadings to make it more clear

File: txmatching/web/api/user_api.py (hunk 5 above)
Comment (NotLubo) [reply txmatching/web/api/user_api.py:184]: Done

File: README.md (hunk 1 above)
Comment (NotLubo) [reply README.md]: Done

**Comment by kristinagalik** [comment]
@NotLubo 

potom pre info: 
```
    ports:
      - "9000:9000"
      - "9443:9443"
    extra_hosts:
      - "host.docker.internal:host-gateway"
```

toto mi tiez neslo:
image

**Review by kubantjan — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -124,3 +124,32 @@ To obtain configuration in the code, one should call `get_application_configurat
 
 ### Testing
 To run unittests use `make test` command. `make check` command runs linter and unittests altogether.
+
+### Authentik
+We are using Authentik for authentication. You can find more information about it [here](https://goauthentik.io/docs).
+After you run `make docker-build` you must setup Authentik. You will do this only once.
```
Comment (kubantjan) [inline README.md:130]: there is no need for docker-build you need only docker-compose up! and make docker-build does not even do that it just build txmatch image that is not related at all

File: docker-compose.yml [hunk 7]
```diff
@@ -24,5 +26,60 @@ services:
     volumes:
       - txmatching-postgres:/var/lib/postgresql/data/
 
+  postgresql:
```
Comment (kubantjan) [inline docker-compose.yml:29]: authentik-db or some clear name like that

File: docker-compose.yml [hunk 8]
```diff
@@ -24,5 +26,60 @@ services:
...
+    image: postgres:12-alpine
+    restart: unless-stopped
+    volumes:
+      - database:/var/lib/postgresql/data
+    environment:
+      - POSTGRES_PASSWORD=${PG_PASS:-Your_password123}
+      - POSTGRES_USER=${PG_USER:-admin}
+      - POSTGRES_DB=${PG_DB:-authentik}
+  redis:
```
Comment (kubantjan) [inline docker-compose.yml:38]: authentik-redis?

File: docker-compose.yml [hunk 9]
```diff
@@ -24,5 +26,60 @@ services:
...
+    image: redis:alpine
+    restart: unless-stopped
+  server:
```
Comment (kubantjan) [inline docker-compose.yml:41]: authentik-server?

File: docker-compose.yml [hunk 10]
```diff
@@ -24,5 +26,60 @@ services:
...
+      AUTHENTIK_POSTGRESQL__PASSWORD: ${PG_PASS:-Your_password123}
+      # AUTHENTIK_ERROR_REPORTING__ENABLED: "true"
+      # WORKERS: 2
+    volumes:
+      - ./media:/media
+      - ./custom-templates:/templates
+    ports:
+      - "9000:9000"
+      - "9443:9443"
+  worker:
```
Comment (kubantjan) [inline docker-compose.yml:61]: authentik-worker?

**Review by kubantjan — CHANGES_REQUESTED** [review]
you are missing depends_on in the docker-compose services. Use that. to clarify which service is needed for which other service

File: docker-compose.yml [hunk 11]
```diff
@@ -24,5 +26,60 @@ services:
...
+      - database:/var/lib/postgresql/data
+    environment:
+      - POSTGRES_PASSWORD=${PG_PASS:-Your_password123}
+      - POSTGRES_USER=${PG_USER:-admin}
+      - POSTGRES_DB=${PG_DB:-authentik}
+  redis:
+    image: redis:alpine
+    restart: unless-stopped
+  server:
+    image: ${AUTHENTIK_IMAGE:-ghcr.io/goauthentik/server}:${AUTHENTIK_TAG:-2022.4.1}
```
Comment (kubantjan) [inline docker-compose.yml:42]: k cemu je tady ten AUTHENTIK_IMAGE env?  to asi k nicemu nikdy neni ne?

Comment (kubantjan) [reply docker-compose.yml:42]: Klidne bych tady smazal i ten ENV AUTHENTIK_TAG

**Review by kubantjan — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 11 above)
Comment (NotLubo) [reply docker-compose.yml:42]: Myslel jsem si, že pak držíš verzi authentiku v env ale mohu zmazat

File: docker-compose.yml [hunk 12]
```diff
@@ -24,5 +26,60 @@ services:
...
+    environment:
+      AUTHENTIK_SECRET_KEY: DdP75bZYo8eOfHyCUL2bL5dFX5O5WQy6p5afJrASFpk6hlWW3c
+      AK_ADMIN_TOKEN: LHDXJc2o55Zed6YP5Wd57CUpSLOr36AOypbhfQW5Fylb8kaFfW
+      AUTHENTIK_REDIS__HOST: redis
+      AUTHENTIK_POSTGRESQL__HOST: postgresql
+      AUTHENTIK_POSTGRESQL__USER: ${PG_USER:-admin}
+      AUTHENTIK_POSTGRESQL__NAME: ${PG_DB:-authentik}
+      AUTHENTIK_POSTGRESQL__PASSWORD: ${PG_PASS:-Your_password123}
+      # AUTHENTIK_ERROR_REPORTING__ENABLED: "true"
+    # This is optional, and can be removed. If you remove this, the following will happen
```
[the remark this answers is not in the document]
Comment (NotLubo) [reply docker-compose.yml:74]: I think the comment belongs to something that is already removed. I asked Honza.

[the remark this answers is not in the document]
Comment (NotLubo) [reply docker-compose.yml:74]: the comment is meant for `user: root`. I've corrected it

File: docker-compose.yml (hunk 11 above)
Comment (kubantjan) [reply docker-compose.yml]: Asi bych to smazal v dohledné době to nepoužijeme přes ten env

Comment (NotLubo) [reply docker-compose.yml]: Je to zmazane

**Review by kubantjan — APPROVED** [review]

**Pull request merged**